### DISCUSSION TITLE
Commit message "feature: :sparkles: something" doesn't show ✨ but "feature: something :sparkles:" and "feature: ✨ something" does.

### DISCUSSION THREAD

**Discussion by JHolcman-T** [community/community#78236] [Repositories]
### Select Topic Area

Bug

### Body

Hi,

topic like in title. Is this intended *"feature"* or **bug**?

I would like to add my emojis (*using colon syntax*) first and then the rest of the commit message.

For now workaround with adding UTF-8 char for ✨ works.

**Comment by erickdc7** [comment]
Hello,

The behavior you're experiencing with emoji rendering in commit messages on GitHub is likely due to the way the platform parses and displays emojis. When using colon syntax for emojis, such as `:sparkles:`, GitHub converts it into the corresponding emoji graphic. However, this conversion might not always work as expected, especially when the emoji is placed at the beginning of the commit message.

This could be related to how GitHub's markdown rendering engine interprets the text. It's possible that placing the colon syntax at the beginning of the message may not be recognized as an emoji, while placing it later in the message or using the actual UTF-8 character is recognized and rendered correctly.

As you've found, using the actual UTF-8 emoji character is a reliable workaround. Another option is to consult resources like [gitmoji](https://gitmoji.dev/), which provides a guide for using emojis in commit messages, to see if there are any specific recommendations or best practices for emoji placement.

If you believe this is a bug or an unintended feature, you can report it to GitHub's support or issue tracker, providing details of your observations and any steps to reproduce the issue. They may be able to provide further insights or address the behavior in future updates.

For now, continuing to use the UTF-8 character directly in your commit messages seems to be the best approach to ensure consistent rendering of emojis.

**Reply by GabrielLins64** [reply]
Yes, this issue stems from the platform's parsing mechanism. However, this behavior wasn't default in GitHub previously. It disrupts older commit messages following the pattern '_something_: :some_emoji:'. Previously, these emojis used to render correctly but stopped functioning about 2 or 3 weeks ago. In my opinion, this **does constitutes a bug** that needs addressing as it interferes with widely used extensions like [Conventional Commits](https://marketplace.visualstudio.com/items?itemName=vivaxy.vscode-conventional-commits) and breaks old commit messages which used to display correctly.

Check this discussion out: https://github.com/orgs/community/discussions/77606

**Comment by burhankhanzada** [comment]
I also noticed :emoji: doesn't render in commit message but it does before.
